pkg/cache/sql/db/client.go

```diff
@@ -320,6 +324,15 @@ func (c *Client) NewConnection() error {
 		return err
 	}
 
+	// Touch the file first which means that when SQLite opens it, it will
+	// retain the permissions.
+	// This is done because we can't control the permissions via the sql.Open
```

Comment:
This comment could be clearer. Maybe something like:

Set the permissions in advance, because we can't control them if
the file is created by a `sql.Open` call instead.